File: 백준/Gold/2467. 용액/용액.py
import sys;
from collections import deque
input = sys.stdin.readline

n = int(input())
lst = list(map(int,input().split()))
sp, ep = 0, n-1
Min = 2000000001
result = []

while sp < ep :
    value = lst[sp] + lst[ep]

    if abs(value) < Min :
        Min = abs(value)
        result = [lst[sp],lst[ep]]

    if value <= 0 :
        sp += 1

    elif value > 0 :
        ep -= 1

print(*result)

File: 백준/Gold/2467. 용액/용액.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const fs = require("fs")
let input = fs.readFileSync(filePath).toString().trim().split("\n")

const [n] = input[0].split(" ").map((e) => Number(e))
const nums = input[1].split(" ").map((e) => Number(e))

let sp = 0, ep = n - 1
let zero = 2000000001
let result = []

while(sp < ep) {
  const value = nums[sp] + nums[ep]
  
  if (zero > Math.abs(value)){
    zero = Math.abs(value)
    result = [nums[sp] , nums[ep]]
  } 

  if (value <= 0){
    sp++
  }
  
  else if (value > 0){
    ep--
  }
}

console.log(...result)